Update searchable snapshot doc about the timing to notice data loss (#112050)
elastic/elasticsearch: docs/reference/searchable-snapshots/index.asciidoc

delete these snapshots. The snapshot contains the sole full copy of your data.
If you delete it then the data cannot be recovered from elsewhere.

> * If snapshot data are cached onto local storage, your cluster may remain the functionality while the cache is alive. However, for partially mounted indices, the cache is cleared when a node is restarted. Therefore, if you delete the searchable snapshot, it's possible that you may notice the data loss later, for example, during a node restart. 

Reviewer: ```suggestion
* If snapshot data are cached in local storage, your cluster may retain the functionality while the cache is alive. However, for partially mounted indices, the cache is cleared when a node is restarted. This means that if you delete a searchable snapshot, you might not notice the data loss until a node restart.
```

Reviewer 2: I think we can tighten up the language here. It's not a question of "if" the data is cached - the data is always cached locally. And this doesn't only apply to partially cached shards so let's drop the mention of that

Reviewer: Updated!

Reviewer 2: How about this?

```suggestion
* The data in a searchable snapshot index are cached in local storage, so if you
delete the underlying searchable snapshot {es} will continue to operate normally
until the first cache miss. This may be much later, for instance when a shard
relocates to a different node or a node restarts.
```
